pkgs/development/idris-modules/with-packages.nix

```diff
@@ -1,46 +1,20 @@
 # Build a version of idris with a set of packages visible
 # packages: The packages visible to idris
-{ stdenv, idris }: packages: stdenv.mkDerivation {
-  inherit (idris) name;
+{ stdenv, idris, symlinkJoin, makeWrapper }: packages:
 
-  buildInputs = packages;
+let paths = stdenv.lib.closePropagation packages;
+in
+symlinkJoin {
 
-  preHook = ''
-    mkdir -p $out/lib/${idris.name}
+  name = idris.name + "-with-packages";
 
-    installIdrisLib () {
-      if [ -d $1/lib/${idris.name} ]; then
-        ln -fsv $1/lib/${idris.name}/* $out/lib/${idris.name}
-      fi
-    }
+  paths = paths ++ [idris] ;
 
-    envHooks+=(installIdrisLib)
-  '';
+  buildInputs = [ makeWrapper ];
 
-  unpackPhase = ''
-    cat >idris.c <<EOF
-    #include <stdlib.h>
-    #include <unistd.h>
-    #include <stdio.h>
+  postBuild = ''
+    wrapProgram $out/bin/idris \
+      --set IDRIS_LIBRARY_PATH $out/libs
```

Comment:
Won't this overwrite a user-set `IDRIS_LIBRARY_PATH`?

Reply:
That's a good point. Would a good solution to be to also symlink the already set `IDRIS_LIBRARY_PATH` into the right place?

Reply:
Originally I just had it respect the original `IDRIS_LIBRARY_PATH` and not override it. The real solution is to let IDRIS_LIBRARY_PATH have multiple colon-separated entries though.

Reply:
I don't think it is terrible to overwrite the `IDRIS_LIBRARY_PATH` for now, users can easily package their libraries and let nix manage them.